src/Livewire/Features/Calendar/FluxCalendar.php

```diff
@@ -31,6 +32,73 @@ class FluxCalendar extends CalendarComponent
 
     protected string $view = 'flux::livewire.features.calendar.flux-calendar';
 
+    protected $listeners = [
+        'calendar-view-did-mount' => 'storeViewSettings',
+        'calendar-toggle-event-source' => 'toggleEventSource',
+    ];
+
+    protected function getCacheKey(): string
+    {
+        return static::class;
+    }
+
+    #[Renderless]
+    public function toggleEventSource(array ...$activeCalendars): void
+    {
+        $this->storeSettings(array_column($activeCalendars, 'publicId'), 'activeCalendars');
+    }
+
+    #[Renderless]
+    public function storeViewSettings(array $view): void
+    {
+        $this->storeSettings(data_get($view, 'type'), 'initialView');
+    }
+
+    #[Renderless]
+    public function storeSettings(mixed $data, ?string $setPath = null): void
+    {
+        if (! in_array(HasCalendarUserSettings::class, class_uses_recursive(auth()->user()))) {
+            return;
+        }
+
+        $currentData = auth()->user()->getCalendarSettings(static::class)->value('settings') ?? [];
+
+        if (! is_null($setPath)) {
+            data_set($currentData, $setPath, $data);
+            $setData = $currentData;
+        } else {
+            $setData = Arr::undot(
+                array_merge(
+                    Arr::dot($currentData),
+                    Arr::dot($data)
```

Comment:
$data may not something that is iterable. If this is the case, this may throw an exception.